lib/std/static_array_map.zig

```diff
@@ -0,0 +1,452 @@
+    try testing.expectEqual({}, map.get("these").?);
+    try testing.expectEqual({}, map.get("samelen").?);
+
+    try testing.expect(!map.has("missing"));
+    try testing.expect(map.has("these"));
+
+    try testing.expect(null == map.get(""));
+    try testing.expect(null == map.get("averylongstringthathasnomatches"));
+}
+
+fn testStaticStringMapIgnoreCase(comptime map: TestMapIgnoreCase) !void {
+    try testMap(map);
+    try testing.expectEqual(TestEnum.A, map.get("HAVE").?);
+    try testing.expectEqual(TestEnum.E, map.get("SameLen").?);
+    try testing.expect(null == map.get("SameLength"));
+    try testing.expect(map.has("ThESe"));
+}
+
+test "StaticStringMapIgnoreCase" {
+    const slice = [_]TestKV{
+        .{ "these", .D },
+        .{ "have", .A },
+        .{ "nothing", .B },
+        .{ "incommon", .C },
+        .{ "samelen", .E },
+    };
+
+    try testStaticStringMapIgnoreCase(TestMapIgnoreCase.initComptime(slice));
+}
+
+test "empty" {
+    const m1 = StaticStringMap(usize).initComptime(.{});
+    try testing.expect(null == m1.get("anything"));
+
+    const m2 = StaticStringMapIgnoreCase(usize).initComptime(.{});
+    try testing.expect(null == m2.get("anything"));
+}
+
+test "redundant entries" {
+    const slice = [_]TestKV{
+        .{ "redundant", .D },
+        .{ "theNeedle", .A },
+        .{ "redundant", .B },
+        .{ "re" ++ "dundant", .C },
+        .{ "redun" ++ "dant", .E },
+    };
+    const map = TestMap.initComptime(slice);
+
+    // No promises about which one you get:
+    try testing.expect(null != map.get("redundant"));
+
+    // Default map is not case sensitive:
+    try testing.expect(null == map.get("REDUNDANT"));
+
+    try testing.expectEqual(TestEnum.A, map.get("theNeedle").?);
+}
+
+test "redundant insensitive" {
+    const slice = [_]TestKV{
+        .{ "redundant", .D },
+        .{ "theNeedle", .A },
+        .{ "redundanT", .B },
+        .{ "RE" ++ "dundant", .C },
+        .{ "redun" ++ "DANT", .E },
+    };
+
+    const map = TestMapIgnoreCase.initComptime(slice);
+
+    // No promises about which result you'll get ...
+    try testing.expect(null != map.get("REDUNDANT"));
+    try testing.expect(null != map.get("ReDuNdAnT"));
+    try testing.expectEqual(TestEnum.A, map.get("theNeedle").?);
+}
+
+test "comptime-only value" {
+    const map = StaticStringMap(type).initComptime(.{
+        .{ "a", struct {
+            pub const foo = 1;
+        } },
+        .{ "b", struct {
+            pub const foo = 2;
+        } },
+        .{ "c", struct {
+            pub const foo = 3;
+        } },
+    });
+
+    try testing.expect(map.get("a").?.foo == 1);
+    try testing.expect(map.get("b").?.foo == 2);
+    try testing.expect(map.get("c").?.foo == 3);
+    try testing.expect(map.get("d") == null);
+}
+
+test "sorting kvs doesn't exceed eval branch quota" {
+    // from https://github.com/ziglang/zig/issues/19803
+    const TypeToByteSizeLUT = std.StaticStringMap(u32).initComptime(.{
```

Comment:
for consistency, `std.` should be removed.  i think i added this test and forgot to remove the prefix.  this allows way the tests to be run standalone if they're copied elsewhere.

Reply:
Thanks! I also noticed this:
![image](https://github.com/user-attachments/assets/933d0acf-39c5-4962-9c0c-c08fa39ddf8b)
Do you happen to know why these tests are identical? My first instinct was to remove one of them, but I think I'll just change the "slice" test so that it just slices the array.